front/components/assistant_builder/useTemplate.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+  getAgentActionConfigurationType,
+  isDustAppRunConfiguration,
+  isProcessConfiguration,
+  isRetrievalConfiguration,
+  isTablesQueryConfiguration,
+} from "@dust-tt/types";
+import _ from "lodash";
+import { useRouter } from "next/router";
+import { useCallback, useState } from "react";
+
+import type { AssistantBuilderState } from "@app/components/assistant_builder/types";
+import { getDefaultAssistantState } from "@app/components/assistant_builder/types";
+import type { FetchAssistantTemplateResponse } from "@app/pages/api/w/[wId]/assistant/builder/templates/[tId]";
+
+export function useTemplate(
+  initialTemplate: FetchAssistantTemplateResponse | null
+) {
+  const [template, setTemplate] =
+    useState<FetchAssistantTemplateResponse | null>(initialTemplate);
+  const [instructionsResetAt, setInstructionsResetAt] = useState<number | null>(
+    null
+  );
+  const router = useRouter();
+
+  const resetTemplate = useCallback(async () => {
+    setTemplate(null);
```

Comment:
Ok so resetTemplate means "remove template" ?

Reply:
Yes and that's worth renaming 👍🏻 
Adding a commit